app/models/case/base.rb

```diff
@@ -348,7 +348,18 @@ def create!(_attributes = {})
            foreign_key: :case_id
   has_many :related_cases,
            through: :related_case_links,
-           source: :linked_case
+           source: :linked_case,
+           :after_remove => :delete_reverse_links
+
+  has_many :original_case_links,
+           -> { original },
+           class_name: 'LinkedCase',
+           foreign_key: :case_id
+
+  has_many :original_cases,
+           through: :original_case_links,
+           source: :linked_case,
+           :after_remove => :delete_reverse_links
```

Comment:
Do you need the old style arrow syntax here? Could it not just be (same on line 352):

```
after_remove: :delete_reverse_links
```

🤔

Reply:
tried,  doesn't need to use the arrow syntax, I removed them